extmod/modusocket.c

```diff
@@ -324,6 +366,12 @@ STATIC mp_obj_t socket_setsockopt(size_t n_args, const mp_obj_t *args) {
         val = mp_obj_get_int_truncated(args[3]);
         optval = &val;
         optlen = sizeof(val);
+    } else if (opt == 20 && args[3] == mp_const_none) {
+        optval = MP_OBJ_NULL;
+        optlen = 0;
+    } else if (opt == 20 && mp_obj_is_callable(args[3])) {
+        optval = args[3];
+        optlen = sizeof(optval);
```

Comment:
don't you think this should now set the `self->callback` entry, instead of delegating to the implementation?

Reply:
It's possible, but the call may fail for any reason (for example the driver can detect if socket in listen state or if poll queue is full) and it would leave the socket state unchanged.

Reply:
OK, let's leave it as-is then